sim/paladin/item_sets_pve.go

```diff
@@ -123,6 +123,95 @@ var ItemSetLawbringerRadiance = core.NewItemSet(core.ItemSet{
 	},
 })
 
+var ItemSetLawbringerWill = core.NewItemSet(core.ItemSet{
+	Name: "Lawbringer Will",
+	Bonuses: map[int32]core.ApplyEffect{
+		2: func(agent core.Agent) {
+			// (2) Set: Increases the block value of your shield by 30.
+			character := agent.GetCharacter()
+			character.AddStat(stats.BlockValue, 30)
+		},
+		4: func(agent core.Agent) {
+			// (4) Set: Heal for 189 to 211 when you Block. (ICD: 3.5s)
+			// Note: The heal does not scale with healing/spell power, but can crit.
+			paladin := agent.(PaladinAgent).GetPaladin()
+			c := agent.GetCharacter()
+			actionID := core.ActionID{SpellID: 456540}
+
+			bastionOfLight := paladin.RegisterSpell(core.SpellConfig{
+				ActionID:         actionID,
+				SpellSchool:      core.SpellSchoolHoly,
+				DefenseType:      core.DefenseTypeMagic,
+				ProcMask:         core.ProcMaskSpellHealing,
+				DamageMultiplier: 1,
+				ThreatMultiplier: 1,
+				ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+					baseHeal := sim.Roll(189, 211)
+					spell.CalcAndDealHealing(sim, target, baseHeal, spell.OutcomeHealingCrit)
+				},
+			})
+
+			handler := func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				bastionOfLight.Cast(sim, result.Target)
+			}
+
+			core.MakeProcTriggerAura(&c.Unit, core.ProcTrigger{
+				ActionID:   actionID,
+				Name:       "S03 - Item - T1 - Paladin - Protection 4P Bonus",
+				Callback:   core.CallbackOnSpellHitTaken,
+				Outcome:    core.OutcomeBlock,
+				ProcChance: 1.0,
+				ICD:        time.Millisecond * 3500,
+				Handler:    handler,
+			})
+		},
+		6: func(agent core.Agent) {
+
+			paladin := agent.(PaladinAgent).GetPaladin()
+
+			paladin.RegisterAura(core.Aura{
+				Label: "S03 - Item - T1 - Paladin - Protection 6P Bonus",
+				OnReset: func(aura *core.Aura, sim *core.Simulation) {
+					auras := paladin.holyShieldAura
+					procs := paladin.holyShieldProc
+					blockBonus := 30.0 * core.BlockRatingPerBlockChance
+
+					for i, values := range HolyShieldValues {
+
+						if paladin.Level < values.level {
+							break
+						}
+
+						damage := values.damage
+
+						// Holy Shield's damage is increased by 100% of shield block value.
+						procs[i].ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+							sbv := paladin.BlockValue()
+							// Reminder: Holy Shield can crit, but does not miss.
+							spell.CalcAndDealDamage(sim, target, (damage + sbv), spell.OutcomeMagicCrit)
+						}
+
+						// Holy Shield aura no longer has stacks...
+						auras[i].MaxStacks = 0
+
+						// ...and does not set stacks on gain...
+						auras[i].OnGain = func(aura *core.Aura, sim *core.Simulation) {
+							paladin.AddStatDynamic(sim, stats.Block, blockBonus)
+						}
+
+						// ...or remove stacks on block.
+						auras[i].OnSpellHitTaken = func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
```

Comment:
You should be able to put all of this inside an `OnInit` instead of an `OnReset` so that it will run once before sims start and not have to be re-run on each iteration